ayaneo-platform.c

```diff
@@ -927,11 +966,391 @@ struct led_classdev_mc ayaneo_led_mc = {
+ *  depending on the POWER_SUPPLY_PROP_CHARGE_CONTROL_START_THRESHOLD,
+ *  POWER_SUPPLY_PROP_CHARGE_CONTROL_END_THRESHOLD and POWER_SUPPLY_PROP_CHARGE_TYPE.
+ *
+ *  When the writer thread begins its next loop, it checks if it should activate the
+ *  bypass charge if POWER_SUPPLY_CHARGE_TYPE_CUSTOM is set. Then it probes the battery
+ *  charge level to determine if battery charging is still allowed. It then saves the
+ *  state of charge type and sleeps for some seconds.
+ *
+ *  During suspend kthread_stop is called which causes the writer thread to
+ *  terminate after its current iteration. The writer thread is restarted during
+ *  resume to allow updates to continue.
+ */
+static struct task_struct *ayaneo_bypass_charge_writer_thread;
+int ayaneo_bypass_charge_writer(void *pv);
+int ayaneo_bypass_charge_writer(void *pv)
+{
+        char battery[]= "BAT0";
+        int ret = 0;
+        struct power_supply *psy = power_supply_get_by_name(battery);
+        union power_supply_propval capacity;
+        u8 last_charge_type = 0xff;
+        if(psy) {
+            pr_info("Bypass-Writer thread started.\n");
+
+            while (!kthread_should_stop())
+            {
+                if(ps_priv.charge_types == POWER_SUPPLY_CHARGE_TYPE_CUSTOM) {
+                    ret = power_supply_get_property(psy, POWER_SUPPLY_PROP_CAPACITY, &capacity);
+                    if(!ret) {
+                        if(capacity.intval >= ps_priv.current_end_threshold) {
+                            switch (model) {
+                                case air:
+                                case air_1s:
+                                case air_1s_limited:
+                                case air_pro:
+                                case air_plus_mendo:
+                                case geek:
+                                case geek_1s:
+                                case ayaneo_2:
+                                case ayaneo_2s:
+                                case kun:
+                                        ayaneo_bypass_charge_legacy_open();
+                                        break;
+                                case air_plus:
+                                case slide:
+                                        ayaneo_bypass_charge_open();
+                                        break;
+                                default:
+                                        break;
+                            }
+                        } else if(capacity.intval < ps_priv.current_start_threshold) {
+                            switch (model) {
+                                case air:
+                                case air_1s:
+                                case air_1s_limited:
+                                case air_pro:
+                                case air_plus_mendo:
+                                case geek:
+                                case geek_1s:
+                                case ayaneo_2:
```

Comment:
The 2 & geek don't have a BIOS that sorts this, we should omit anything we haven't confirmed for safety.

Reply:
This would be better.